submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,211 @@
+
+![Picture of my first CPU design](./img/mk1_computer.png "Ah, Factorio. One of the only games where you literally get smarter on repeated playthroughs.")
+
+Ironically, I'm not actually here to talk about *any* of this. This is all just foreshadowing.
+
+I was starting the second revision of the computer and I wanted a more compact ROM. I came up with the design below, which has the capability of storing true 32-bit numbers and can store 4 KiB of data per row:
+
+![Depiction of my ROM](./img/ROM_design.png "Everything starts somewhere.")
+
+The ROM is *very* dense, but it functions on a system where each value is split into and stored as two 16-bit numbers, which are then recombined on output. ROMs are normally tedious to create, manually encoding each signal you want by hand, one at a time, but this design was even more involved. I now had to split, bitwise AND, bitwise right-shift, and populate not one signal, but two, in two different specific locations, making sure that they both had the correct value *and* signal type. Needless to say, it was going to be painful to set tens of values, let alone hundreds, or the *thousands* the machine was capable of storing.
+
+The solution? Get a computer to do it for me. It can stomach this task far better than I ever could, and can do it way faster as well. Factorio's blueprint string import function can take any correctly formatted string; all I had to do was create this text string to my specification with the data I wanted and I could just paste it where needed.
+
+This concept is not new, at all. Even a cursory search will turn up numerous examples of this used in practice: [`factorio-blueprint` NPM module by demipixel](https://github.com/demipixel/factorio-blueprint), [justarandomgeek's compiler for his massive combinator computer](https://github.com/justarandomgeek/factorio-computer), [a generic combinator instruction language by Jobarion](https://github.com/Jobarion/factoriogen/), [image-to-blueprint converters](https://github.com/EFLFE/Artorio), etc. The list is extensive.
+
+With all these examples, I was hoping that I could find some similarities with one of them and use that as a baseline for my solution to avoid "retreading" the same ground. Something troubled me about the solutions that were available though: They all had problems!
+
+* Many implementations were highly specific to the domain they were written for; a combinator computer compiler script was not going to be portable for many other purposes than compiling code for a specific combinator computer.
+* They lacked a unified language; many were written in Lua, some in Python, another in JavaScript, one in C++, etc. This meant that each one had to write its own implementations for the same operations, instead of just having someone write the implementation they need and making it available to others.
```

Comment:
```suggestion
* They lacked a unified language; many were written in Lua, some in Python, another in JavaScript, one in C++, etc. This meant that each one had to write their own implementations for the same operations, instead of just having someone write the implementation they need and making it available to others.
```